tools/run_asan.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -e
+set -x
+
+BUILDDIR="_leakcheckbuild"
+
+meson setup ${BUILDDIR} -Db_sanitize=address,undefined
+meson compile -C ${BUILDDIR} --verbose
+
+TESTS=$(find ${BUILDDIR}/tests/ctests/ -executable -type f)
+for test in ${TESTS}
+do
+    ./${test}
+done
+
+mkdir -p ${BUILDDIR}/fakeroot/{etc/netplan,run}
+export LD_LIBRARY_PATH="${BUILDDIR}/src"
+
+for yaml in examples/*.yaml
+do
+    chmod 600 ${yaml}
```

Comment:
We shouldn't be modifying the file permissions inside the source directory, but rather first copy it over into the fakeroot and change permissions there before execution of `generate`